examples/2-query/src/Monster.re

```diff
@@ -0,0 +1,110 @@
+open MonsterStyles;
+open ReasonUrql;
+
+type maximum = {maximum: int};
+
+type pokemon = {
+  id: string,
+  number: string,
+  height: maximum,
+  weight: maximum,
+};
+
+/* graphql query for GetPokemon */
+
+module GetPokemon = [%graphql
+  {|
+  query pokemon($name: String!) {
+    pokemon(name: $name) {
+        name
+        classification
+        height {
+            maximum
+        }
+        weight {
+            maximum
+        }
+      image
+    }
+  }
+|}
+];
+let component = ReasonReact.statelessComponent("Example");
+
+let make = (~pokemon: string, _children) => {
+  ...component,
+  render: _self => {
+    /* We set up the query here as we need access to the pokemon
+       value passed in from GetAll */
+    let query = GetPokemon.make(~name=pokemon, ())##query;
+    let variables = Js.Dict.empty();
```

Comment:
So rather than needing to encode the variables yourself, `graphql_ppx` encodes them for you and allows you to access them as `##variables`. So you can do something like the following:

```reason
let request = GetPokemon.make(~name=pokemon, ());
let query = request##query;
let variables = request##variables; /* this will be a Js.Json.t */
```

Reply:
Ah this explains a question I had about having to redefine variables a couple of times!